### PR TITLE
Fix project discovery for large projects with multiple packages

### LINKED ISSUE #830: High CPU usage causing VS Code to be unstable
### Expected behavior

Normal CPU Usage

### Actual behavior

After launching VS Code there is a warning dialog with the message: 
"dbt Power User detected 63 projects in your work space, this will negatively affect performance."

I suspect that the extension is detecting all the packages that my project uses in the dbt_packages folder as individual projects. 
My project has been like this for some time however, and I have never seen this warning message before.

### Steps To Reproduce

Open VS Code, open a folder containing a dbt project.

I suspect that the extension is detecting all the packages that my project uses in the dbt_packages folder as individual projects. 

### Log output/Screenshots

Screenshot 2024-01-17 at 11 35 09

### Operating System

Ubuntu 20.04

### dbt version

1.4.9

### dbt Adapter

Snowflake

### dbt Power User version

0.29.1

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### PR DESCRIPTION
**Pull request by mdesmet**
## Overview
resolves #830 

## Checklist

- [ ] I have run this code and it appears to resolve the stated issue
- [ ] `README.md` updated and added information about my change

### CODE REVIEW

File: dbt_integration.py [hunk 1]
```diff
@@ -168,6 +169,26 @@ def wrapper(
 
     return wrapper
 
+def default_profiles_dir(project_dir) -> Path:
+    if "DBT_PROFILES_DIR" in os.environ:
+        return Path(os.environ["DBT_PROFILES_DIR"]).resolve()
+    return project_dir if (project_dir / "profiles.yml").exists() else Path.home() / ".dbt"
```
Comment (suryaiyer95) [inline dbt_integration.py:175]: os.path.join() would make it is agnostic

Comment (mdesmet) [reply dbt_integration.py:175]: it is agnostic already as we use vscode.Uri later on.

File: dbt_integration.py [hunk 2]
```diff
@@ -168,6 +169,26 @@ def wrapper(
...
+
+def find_package_paths(project_directories, profiles_dir_override):
+    def get_package_path(project_dir):
+        try:
+            project = DbtProject(project_dir=project_dir, profiles_dir=profiles_dir_override if Path(profiles_dir_override).exists() and profiles_dir_override.strip() != ""  else default_profiles_dir(Path(project_dir)))
+            project.init_config()
+            packages_path = Path(project.config.packages_install_path)
+            if packages_path.is_absolute():
+                return packages_path.resolve().as_uri()
+            return (Path(project_dir) / packages_path).as_uri()
```
Comment (suryaiyer95) [inline dbt_integration.py:185]: os.path.join

**Review by suryaiyer95 — COMMENTED** [review]
Minor comment

**Review by anandgupta42 — APPROVED** [review]

**Pull request merged**